app/src/hid_listener.c

```diff
@@ -70,17 +71,81 @@ static int hid_listener_keycode_released(const struct zmk_keycode_state_changed
     return zmk_endpoints_send_report(ev->usage_page);
 }
 
+static void zmk_mouse_work(struct k_work *work) {
+    int rc = zmk_endpoints_send_mouse_report();
+    if (rc != 0) {
+        LOG_ERR("Failed to send mouse report, error: %d", rc);
+    }
+}
+
+K_WORK_DEFINE(mouse_work, &zmk_mouse_work);
+
+void mouse_timer_cb(struct k_timer *dummy) { k_work_submit(&mouse_work); }
+
+K_TIMER_DEFINE(mouse_timer, mouse_timer_cb, mouse_timer_cb);
+
+static int mouse_timer_ref_count = 0;
+
+void mouse_timer_ref() {
+    if (mouse_timer_ref_count == 0) {
+        k_timer_start(&mouse_timer, K_NO_WAIT, K_MSEC(10));
+    }
+    mouse_timer_ref_count += 1;
+}
+
+void mouse_timer_unref() {
+    mouse_timer_ref_count -= 1;
+    if (mouse_timer_ref_count == 0) {
+        k_timer_stop(&mouse_timer);
+    }
+}
+
+static int hid_listener_mouse_pressed(const struct zmk_mouse_state_changed *ev) {
+    int err;
+    LOG_DBG("x: 0x%02X, y: 0x%02X", ev->x, ev->y);
+    err = zmk_hid_mouse_movement_press(ev->x, ev->y);
+    if (err) {
+        LOG_ERR("Unable to press button");
+        return err;
+    }
+    mouse_timer_ref();
+    return 0;
+}
+
+static int hid_listener_mouse_released(const struct zmk_mouse_state_changed *ev) {
+    int err;
+    LOG_DBG("x: 0x%02X, y: 0x%02X", ev->x, ev->y);
+    err = zmk_hid_mouse_movement_release(ev->x, ev->y);
+    if (err) {
+        LOG_ERR("Unable to release button");
+        return err;
+    }
+    mouse_timer_unref();
```

Comment:
if the `hid_mouse_movement_release` throws an error, shouldn't the mouse timer unref be called anyway?